### PR TITLE
add github actions for ci

### PR DESCRIPTION
**Pull request by Scanteianu**
use basic starter template from https://github.com/actions/starter-workflows/blob/main/ci/python-app.ym to install requirements and run pytest tests

### CODE REVIEW

File: .github/workflows/github-actions.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
```
Comment (gdams) [inline .github/workflows/github-actions.yml:22]: please pin these to the actual sha

Comment (gdams) [reply .github/workflows/github-actions.yml:22]: e.g `actions/checkout@1d96c772d19495a3b5c517cd2bc0cb401ea0529f #v4.1.3`

Comment (Scanteianu) [reply .github/workflows/github-actions.yml:22]: where can i find the actual shas?

**Comment by Scanteianu** [comment]
> Please also add a dependabot config to keep the github actions dependencies up to date

Thanks for the review! I think i did both of the things

File: .github/workflows/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Set update schedule for GitHub Actions
```
Comment (gdams) [inline .github/workflows/dependabot.yml:1]: I think that this is meant to be in `.github/dependabot.yml`

File: .github/workflows/github-actions.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@37b082107ba410260a3aaddf93122e04801ce631
```
Comment (gdams) [inline .github/workflows/github-actions.yml:22]: Can you add the version as a comment after the SHA for ease of editing (e.g `# v1.0.0`)

File: .github/workflows/dependabot.yml (hunk 2 above)
Comment (Scanteianu) [reply .github/workflows/dependabot.yml:1]: https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot#example-dependabotyml-file-for-github-actions i think i misread this, will move it

File: .github/workflows/github-actions.yml (hunk 3 above)
Comment (Scanteianu) [reply .github/workflows/github-actions.yml]: yes, also changed sha to latest release for both

Comment (Scanteianu) [reply .github/workflows/github-actions.yml]: (as opposed to latest commit with passing checks)

**Review by gdams — APPROVED** [review]

**Review by karianna — APPROVED** [review]

**Pull request merged**